src/include/test/unit_flat_qv.cc

```diff
@@ -220,14 +220,59 @@ TEST_CASE(
   auto array_groundtruth = tdbColMajorPreLoadMatrix<siftsmall_groundtruth_type>(
       ctx, siftsmall_groundtruth_uri);
 
-  auto&& [D00, I00] =
-      detail::flat::qv_query_heap(array_inputs, array_queries, k_nn, 1);
+  SECTION("Invoke with default distance function") {
+    auto&& [D00, I00] =
+        detail::flat::qv_query_heap(array_inputs, array_queries, k_nn, 1);
 
-  size_t intersections00 =
-      (long)count_intersections(I00, array_groundtruth, k_nn);
-  CHECK(intersections00 != 0);
-  size_t expected00 = array_groundtruth.num_cols() * k_nn;
-  CHECK(intersections00 == expected00);
+    size_t intersections00 =
+        (long)count_intersections(I00, array_groundtruth, k_nn);
+    CHECK(intersections00 != 0);
+    size_t expected00 = array_groundtruth.num_cols() * k_nn;
+    CHECK(intersections00 == expected00);
+  }
+
+  SECTION("Invoke with l2_distance function") {
```

Comment:
It would be nice to, but we don't have any ground truth for the other distance functions.  We should also determine what all of the ramifications are for having different distance functions.  I.e., since kmeans can take a distance function, that means the partitioning is going to be different.  That should probably be indicated somehow in the index itself.

Reply:
I don't quite follow, why can't we test with some simple data and just calculate what the results should be?

Reply:
Talked offline: yes, we can and should add tests here.

Reply:
Added a small simple test in https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/265/commits/9847c4efb3920c306ee611f211a46eea718a8bb1